src/sos/preview.py

```diff
@@ -230,11 +230,45 @@ def preview_md(filename, kernel=None, style=None):
     return {'text/html': html, 'text/plain': text}
 
 def preview_dot(filename, kernel=None, style=None):
+    from os import listdir, remove
+    from os.path import isfile, join
     from graphviz import Source
+    import tempfile
+    import shutil
     with open(filename) as dot:
-        src = Source(dot.read())
-    src.format='png'
-    outfile = src.render()
-    with open(outfile, 'rb') as content:
-        data = content.read()
-    return {'image/png': base64.b64encode(data).decode('ascii') }
+        fileNameElement = "sosDotFilesPng"
+        with tempfile.TemporaryDirectory() as tempDirectory:
+            src = Source(dot.read(), filename = fileNameElement, directory = tempDirectory)
+            src.format = 'png'
+            outfile = src.render(filename = fileNameElement, directory = tempDirectory)
+            pngFiles = [f for f in listdir(tempDirectory) if isfile(join(tempDirectory, f)) 
+                        and fileNameElement in f 
+                        and ".png" in f 
+                        and any(x.isdigit() for x in f) 
+                        and fileNameElement in f]
+            try:
+                #if len(pngFiles)==0:
+                with open(outfile, 'rb') as content:
+                    data = content.read()
+                result = {'image/png': base64.b64encode(data).decode('ascii') }
+                if len(pngFiles)!=0:
+                    import imageio
+                    pngFiles.sort(key=lambda x: int(x.split('.')[1]))
+                    pngFiles.insert(0, join(tempDirectory, fileNameElement + '.png'))
+                    images = []
```

Comment:
learn to do something like
```
images = [imageio.imread(.....x) for x in pngFiles]
```

Reply:
Starting to find the list comprehensive style is very convenient!